crypto/bio/bss_sock.c

```diff
@@ -108,10 +126,31 @@ static int sock_read(BIO *b, char *out, int outl)
 
 static int sock_write(BIO *b, const char *in, int inl)
 {
-    int ret;
+    int ret = 0;
 
     clear_socket_error();
-    ret = writesocket(b->num, in, inl);
+    if (BIO_should_ktls_ctrl_msg_flag(b)) {
```

Comment:
It is not mandatory, because all the BIO flag setting/checking is outside of OPENSSL_NO_KTLS. This is comfortable because there are no OPENSSL_NO_KTLS sections in ssl/record/rec_layer_s3.c.
It works because the ktls flags could be set only if ktls is enabled.

Having said that, after removing all the debug prints I've moved it inside the OPENSSL_NO_KTLS section. It is a bit ugly due to the indentation required by the else clause.